src/test/java/uk/gov/hmcts/reform/pip/account/management/service/AccountServiceTest.java

```diff
@@ -507,7 +508,23 @@ void testUploadMediaFromInvalidCsv() throws IOException {
                                                 "Should throw CsvParseException");
             assertTrue(ex.getMessage().contains("Failed to parse CSV File due to"), MESSAGES_MATCH);
 
-
         }
     }
+
+    @Test
+    void testMiEndpoint() {
+        String testString = accountService.getAccManDataForMiReporting();
+        String[] splitLineString = testString.split("\r\n|\r|\n");
+        long countLine1 = splitLineString[0].chars().filter(character -> character == ',').count();
+        assertThat(testString)
+            .as("Header row missing")
+            .contains("provenance_user_id");
+        assertThat(splitLineString)
+            .as("Only one line exists - data must be missing, as only headers are printing")
```

Comment:
The error message says you need to have more than one line, but then you are checking it has size of one or more in "hasSizeGreaterThanOrEqualTo(1)". I wonder you should just mock the returned value from the repository, then check the service class method returns the expected result as the mocked repository returned value. Then add a proper integration test to end the endpoint with real values.

Reply:
That's the next step to get more sonar coverage anyway :p

Reply:
I think it's a good idea to add an integration test once I've had a look at what staging endpoints return (my local pg docker container is very old and probably has some quirks)

Reply:
For this test here, guess you can simply do a test like below, but it is up to you :)

       String line1 = "123, 456, PI_AAD, VERIFIED";
        String line2 = "789, 999, CFT_IDAM, VERIFIED";

        when(userRepository.getAccManDataForMI())
            .thenReturn(List.of(line1, line2));

        assertThat(accountService.getAccManDataForMiReporting().split(System.lineSeparator()))
            .hasSize(3)
            .containsExactly("user_id, provenance_user_id, user_provenance, roles", line1, line2);

Reply:
I've done something like this now - will do the integration test in a later change (probably when the actual request is called somewhere!)